plugins/MultiSites/angularjs/dashboard/dashboard-model.service.js

```diff
@@ -63,9 +63,62 @@
+                    }
+                    if (model.sortColumn == 'pageviews_evolution') {
+                        previousTotal = site.previous_Actions_nb_pageviews;
+                        currentTotal = site.nb_pageviews;
+                        evolution = NumberFormatter.formatPercent(site.pageviews_evolution);
+                        metricName = _pk_translate("General_ColumnPageviews");
+                        previousTotalAdjusted = NumberFormatter.formatNumber(Math.round(site.previous_Actions_nb_pageviews * site.ratio));
+                    }
+                    if (model.sortColumn == 'revenue_evolution') {
+                        previousTotal = NumberFormatter.formatCurrency(site.previous_Goal_revenue, site.currencySymbol);
+                        currentTotal = NumberFormatter.formatCurrency(site.revenue, site.currencySymbol);
+                        evolution = NumberFormatter.formatPercent(site.revenue_evolution);
+                        metricName = _pk_translate("General_ColumnRevenue");
+                        previousTotalAdjusted = NumberFormatter.formatCurrency(Math.round(site.previous_Goal_revenue * site.ratio), site.currencySymbol);
+                    }
+
+                    if (metricName) {
+                        site.tooltip = _pk_translate("MultiSites_EvolutionComparisonIncomplete", [percent]) + "\n";
+                        site.tooltip += _pk_translate("MultiSites_EvolutionComparisonProportional", [percent, previousTotalAdjusted, metricName, previousTotal]) + "\n";
+
+                        switch (site.periodName) {
+                            case 'day':
+                                site.tooltip += _pk_translate("MultiSites_EvolutionComparisonDay", [currentTotal, metricName, previousTotalAdjusted, site.previousRange, evolution]);
+                                break;
+                            case 'week':
+                                site.tooltip += _pk_translate("MultiSites_EvolutionComparisonWeek", [currentTotal, metricName, previousTotalAdjusted, site.previousRange, evolution]);
+                                break;
+                            case 'month':
+                                site.tooltip += _pk_translate("MultiSites_EvolutionComparisonMonth", [currentTotal, metricName, previousTotalAdjusted, site.previousRange, evolution]);
+                                break;
+                            case 'year':
+                                site.tooltip += _pk_translate("MultiSites_EvolutionComparisonYear", [currentTotal, metricName, previousTotalAdjusted, site.previousRange, evolution]);
+                                break;
+                        }
+                    }
+
+                }
+
+                site.visits_evolution_int    = parseInt(site.visits_evolution, 10);
+                site.pageviews_evolution_int = parseInt(site.pageviews_evolution, 10);
+                site.revenue_evolution_int   = parseInt(site.revenue_evolution, 10);
```

Comment:
That actually might not work in various languages. The are languages that are not using a "normal" minus, percent or currency sign. For example: 
https://github.com/matomo-org/matomo/blob/13b22afc4a105262754b51326962c94c305bd6b3/plugins/Intl/lang/sl.json#L878
`parseInt('−234')` returns `NaN`
Same might happen if the currency or percent signs are placed before the number.
Sending a `format_metrics=0` with the request for loading the metrics might not help, as it's overwritten here:
https://github.com/matomo-org/matomo/blob/115527353a9e75e01aa4d263408956ae45403bea/plugins/MultiSites/Dashboard.php#L209
So maybe changing that to something like `$request['format_metrics'] = $request['format_metrics'] ?? 1;` might make that possible.

But that would mean the evolution metrics would be given as floating numbers (eg 0.13 instead of 13%,...)

Reply:
@sgiehl Looking at `format_metrics` gave me the idea to add a new API option `format_metrics_add_raw_copy' which accepts a list of data table columns which will have non-formatted '_raw' copies added to the row before they are formatted. The dashboard-model.service can now request raw unformatted copies of the evolution columns which can be used for up/down arrow decisions instead of parseInt() while still having the main displayed evolution columns properly formatted by the API.

This option is disabled by default and should not incur any overhead unless the request utilizes it. Maybe it could be useful elsewhere when both a formatted and unformatted version of a column are needed by the front end.

Reply:
Seems that approach works. At least some special language edge cases now seem to work:
![image](https://user-images.githubusercontent.com/1579355/140498762-abc4aa19-5394-43ab-9892-fd78e60e90cf.png)

Nevertheless I'm not sure about this new api parameter that allows adding unformatted values. Looks handy in that case, but actually it should also be possible to simply fetch all values unformatted and format them in the UI. Or did you have problems with that?
@tsteur what do you think about this additional api option approach?

Reply:
Turning off formatting in the API would result in unformatted currency values. Since for the multisites API each row is for a different site and may have a different currency, we'd then need to do something like add the currency symbol to each currency column in the datatable at UI level or add a new filter that formats everything except the evolution columns. It just seemed cleaner to let the API keep formatting the datatable values and have optional raw copies of just the required columns, but if that's not considered an acceptable approach then I can try to rework it provide the datatable unformatted and then apply datatable formatting at UI level.

Reply:
Well, you are actually already using `site.currencySymbol` to format some currency values, so the currency for each site already seems to be available.
But yes, we would need to format all metrics in javascript then.
Personally I think the best approach would be to load all metrics unformatted and format them as needed in the UI. We had that problem already a couple of times, that metrics were formatted twice (sometimes even with PHP, e.g. in API and the report rending it). At some point we need to find a way that is clean and easy to understand. And in case we need metrics to calculate with them in javascript we imho should load all metrics unformatted. Otherwise we might end up with having different solutions everywhere.
But maybe @tsteur has another point of view here?

Reply:
I'm not fully understand it all. Generally, if any possible, it be great to have all values coming from the API directly (whether formatted or unformatted). This makes sure we have things formatted consistently the same way and then API consumers like our mobile app also format the same way and show the same data etc. 

Like we could have a column `ratio` and a column `pretty_ratio`. 

If I understand this correctly this logic here generates the tooltip? Initially I was thinking to keep the tooltip simple as I figured it might be taking quite a bit of time. @mattab suggested in https://github.com/matomo-org/matomo/pull/18099#issuecomment-938030176 to maybe use more descriptive tooltips but it's not required if it makes the issue too complicated. I'm not too deep into it. Is there maybe something we can do to get like not the perfect solution but 80% there with little effort? I'm not too much into the problem so I'm not sure what that might be. Maybe it would be printing some numbers unformatted like `1343` instead of `1,343`.

Not sure if that helps?

Reply:
@tsteur the problem actually is, that we try to display a trend picture (positiv / negative / no change) based on the evolution that is received from API. Before there was a "simple" `parseInt` on that number and than it checked if it's positiv, negativ or null. This didn't work properly already before!
While this works for English and a couple of other languages, this won't work for any language that use a different minus sign, or any other format that simply returns `NaN` for parseInt.
The current solution is to have an API parameter `format_metrics_add_raw_copy`, that adds a copy of a certain metric to the response, but unformatted. That works, but might not be the best approach.
Actually this is a general problem we have, whenever we want to calculate in javascript with formatted numbers we received through API.

Reply:
If this was a bug before, we could merge this PR and create a new issue for it? Not sure if I understand it correctly.

> The current solution is to have an API parameter format_metrics_add_raw_copy, that adds a copy of a certain metric to the response, but unformatted.

I'm not much into it. Can we add a new column if trend is positive or negative or no change to the API output? I think this will also help later when there are columns where lower is better etc. Generally it be great when these things come from the API I would say.

Reply:
Adding a new trend column sound reasonable.
@bx80 could you remove the `format_metrics_add_raw_copy` stuff again and instead change the API method to add an additional column for each evolution metric holding a trend value? Guess this should be easily doable using the `ColumnCallbackAddColumn`. Note: filters will be applied before formatting, so there should be no issue with directly checking the evolution values.

Reply:
@sgiehl I don't see what is wrong with the currently implemented and working approach of optionally returning raw metric values from the API where they are needed. Could you help me understand what issues you see with this approach? It's very similar to returning `ratio` and `pretty_ratio` as suggested above and is more generally useful than a positive / negative trend column since it is trivial to determine the trend from the raw figures. Is there a good reason to spend more time on this to achieve a less flexible outcome?

Reply:
@bx80 Sorry, for not being explicit enough. Sometimes I might assume too much that other people would have the same weird thoughts as me 😀 So let me explain that a bit:

Using the raw metrics is something that works in this case. But for other metrics that won't be correct. For some metrics, like page load time, a lower value would actually be better than a higher value, but it wouldn't be possible to see that from the evolution only. So adding some trend value (like `-1`, `0`, `1`) feels like the best approach to me.
There is a method `Metrics::isLowerValueBetter`, that kind of returns if a negative evolution for a column would be a positive trend and wise versa.
So what we actually should imho do is changing the [evolution metric](https://github.com/matomo-org/matomo/blob/115527353a9e75e01aa4d263408956ae45403bea/plugins/CoreHome/Columns/Metrics/EvolutionMetric.php) class to not only add a `_evolution` metric, but also a `_trend` metric. The trend would then be filled based on the evolution and the result of [`Metrics::isLowerValueBetter`](https://github.com/matomo-org/matomo/blob/115527353a9e75e01aa4d263408956ae45403bea/core/Metrics.php#L238-L281) for the wrapped metric. (You can't use that method on the evolution metric directly, as it works for column metrics only).
That way this should be usable in all cases.
This trend value can the be used in the javascript and there shouldn't be any need for the `format_metrics_add_raw_copy` parameter.
Let me know if something is unclear or if you can think of a better solution.

Reply:
@sgiehl Thanks for the detailed explanation, that makes a lot of sense when thinking about metrics where a decreasing value might actually be a good thing! :smiley: I've removed the add raw copy parameter. Instead, when calculating a processed metric, there is a now a check to see if it is an EvolutionMetric, if so then a '_trend' column is added to the datatable with 1, 0 or -1 depending on the calculated evolution and the isLowerValueBetter result. The Javascript then uses the _trend value to decide if up or down arrows are used.